modules/layers/src/layers/path-marker-layer/create-path-markers.ts

```diff
@@ -72,23 +72,26 @@ function createMarkerAlongPath({ path, percentage, lineLength, color, object, pr
   const distanceAlong = lineLength * percentage;
   let currentDistance = 0;
   let previousDistance = 0;
-
-  for (let i = 0; i < path.length - 1; i++) {
+  let i = 0;
+  for (i = 0; i < path.length - 1; i++) {
     currentDistance += path[i].distance(path[i + 1]);
     if (currentDistance > distanceAlong) {
       break;
     }
     previousDistance = currentDistance;
   }
 
-  const lastPoint = path[path.length - 1];
-  const beforeLastPoint = path[path.length - 2];
+  // If reached the end of the loop without exiting early,
+  // undo the final increment to avoid a null-pointer exception
```

Comment:
Why this is needed is better described here: https://github.com/uber/nebula.gl/issues/669